### ISSUE TITLE
Network pop-up not working

### ISSUE THREAD

**Issue by hermidalc** [CBIIT/nci-ctd2-dashboard#280]
*Originally reported by*: @vdancik

Pop-up pages with networks on TGENs Tier 1 submission (http://ctd2-dashboard.nci.nih.gov/#/observation/1775030) are not working. Their pop-up page is an HTML file with a cytoscpe plugin (http://ctd2-dashboard.nci.nih.gov/data/submissions/20150826-tgen_eddy-GBM/GBM_PRIOR_REACTOME0/REACTOME_ABORTIVE_ELONGATION_OF_HIV1_TRANSCRIPT_IN_THE_ABSENCE_OF_TAT/). The page itself is correct as one can copy URL and display page properly in a separate tab/window. 

Can we either:

1. handle plugin/javascript in the pop-up window, or

2. have mechanism to show the network page in its own tab instead?

**Comment by hermidalc** [comment]
*Original comment by*: @vdancik

This was resolved by hosting HTML files at submitters web site. 

A new issue was created for HTML file support (Issue #286)

**Comment by hermidalc** [comment]
The NCI security/app scanner will pick this up as long as it can find the link to it while it's traversing the entire site.  I believe the scanner does traverse every observation page so it should find the link to these files as I can easily traverse to every observation page from links starting on the home page.

On a side note for all the EDDY observations this link is broken:

http://biocomputing.tgen.org/software/EDDY/index

@vdancik should just be this:

http://biocomputing.tgen.org/software/EDDY/

**Comment by hermidalc** [comment]
*Original comment by*: @kcs3

After examining the submission in question, I just want to add a few details.  The submission is organized in a series of directories, one for each piece of network evidence.  Each of the directories contains an index.html file, a qcode.js Javascript file, and a styles.css file.  

The qcode.js file contains an encoded version of the network to be displayed by Cytoscape.js.  qcode.js also contains javascript code to implement the controls you see on the page when it is running.

The index.html file contains links, and also references to a number Javascript scripts hosted on outside servers.  I am told this is normal in ones own website these days.  Here is the relevant part of the code:

    
    <title>EDDY in Cola.js/Cytoscape.js</title>
    
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/css/bootstrap.min.css">
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/css/bootstrap-theme.min.css">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-slider/4.10.3/css/bootstrap-slider.min.css">
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css">
    <link href="http://cdnjs.cloudflare.com/ajax/libs/qtip2/2.2.0/jquery.qtip.min.css" rel="stylesheet" type="text/css" />
    <link href="style.css" rel="stylesheet" />
    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/fastclick/1.0.6/fastclick.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/lodash.js/3.10.0/lodash.min.js"></script>
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.4/jquery.min.js"></script>
    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/js/bootstrap.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-slider/4.10.3/bootstrap-slider.min.js"></script>
    
    <script src="http://marvl.infotech.monash.edu/webcola/cola.v3.min.js"></script>
    <script src="http://cytoscape.github.io/cytoscape.js/api/cytoscape.js-latest/cytoscape.min.js"></script>
    <script src="http://cdnjs.cloudflare.com/ajax/libs/qtip2/2.2.0/jquery.qtip.min.js"></script>
    <script src="https://cdn.rawgit.com/cytoscape/cytoscape.js-qtip/2.2.3/cytoscape-qtip.js"></script>

    <script src="gcode.js"></script>

This content is entered into the submission template as mime type text/html, which is why it opens in the simple pop-up page on top of the evidence page.

Before we can even think about whether to try to make this work (which we have above already basically decided not to do), we need to understand from Leandro (as Vlado already noted) if this submission containing Javascript scripts that will execute in the user's browser is even allowed under NCI rules.  Note that unless the data directory is subjected to security scan this will not be scanned.

**Comment by hermidalc** [comment]
*Original comment by*: @paulclemons

I agree. I think if someone wants to host a service on their OWN server and
link back to it, that's fair game, but we need not to have people expecting
we will install software on the Dashboard server.

**Comment by hermidalc** [comment]
*Original comment by*: @kcs3

I also think that given very limited maintenance resources, we can not really be asked to make Dashboard changes to support anything a submitter happens to come up with.
And even if we did accept such things, it should have been pointed out to the development team for testing before release.

All code is supposed to go through the NCI scanning process.

**Comment by hermidalc** [comment]
*Original comment by*: @vdancik

We may need to check with Leandro whether there is a policy regarding such plugins/scripts.

**Comment by hermidalc** [comment]
*Original comment by*: @kcs3

Do I understand correctly that this submission contains its own executable code (cytoscape plugin)?  This is a beautiful network viewer but it is not clear to me if we should allow submission of outside executable code.  We do have a built-in Cytoscape viewer in the Dashboard but perhaps not as nice as this one.
